test/anchoredinterval.jl

```diff
@@ -583,7 +583,7 @@ using Intervals: Bounded, Ending, Beginning, canonicalize, isunbounded
             r = r[1:5]
 
             # https://github.com/JuliaLang/julia/issues/33882
-            @test r isa StepRangeLen
+            @test r isa (VERSION < v"1.8" ? StepRangeLen : StepRange)
```

Comment:
Best to make these fixes as separate PRs so they can be merged faster

Reply:
After looking into it I think this fix is incorrect. I've made a new fix in #204
```suggestion
            @test r isa StepRangeLen
```